src/pool/srv_pool.c

```diff
@@ -5427,12 +5441,24 @@ evict_iter_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *varg)
 	D_ASSERT(arg->eia_hdl_uuids != NULL);
 	D_ASSERT(arg->eia_hdl_uuids_size > sizeof(uuid_t));
 
-	if (key->iov_len != sizeof(uuid_t) ||
-	    val->iov_len != sizeof(struct pool_hdl)) {
-		D_ERROR("invalid key/value size: key="DF_U64" value="DF_U64"\n",
-			key->iov_len, val->iov_len);
+	if (key->iov_len != sizeof(uuid_t)) {
+		D_ERROR("invalid key size: "DF_U64"\n", key->iov_len);
 		return -DER_IO;
 	}
+	if (val->iov_len != sizeof(struct pool_hdl)) {
+		/* old/2.0 pool handle format ? */
+		if (val->iov_len == sizeof(struct pool_hdl) - 
+		    sizeof(((struct pool_hdl *)0)->ph_machine) &&
+		    arg->eia_pool_svc->ps_global_version < DS_POOL_GLOBAL_VERSION) {
```

Comment:
Should we write "1" instead of DS_POOL_GLOBAL_VERSION, so that when someone changes DS_POOL_GLOBAL_VERSION in the future this condition stays valid? Writing "1" would be accurate anyway, since the format does occur between 1 and 2, and this fact will not change.

Reply:
@wangshilong, what's your expert opinion? ;)

Reply:
Yeah, i agreed.